pulp_rpm/tests/performance/test_publish.py

```diff
@@ -0,0 +1,158 @@
+        2. Assert that repository version is None.
+        3. Sync the remote.
+        4. Assert that repository version is not None.
+        5. Assert that distribution_tree units were added and are present in the repo.
+        6. Publish
+        """
+        delete_orphans(self.cfg)
+        repo = self.client.post(REPO_PATH, gen_repo())
+        self.addCleanup(self.client.delete, repo['pulp_href'])
+
+        # Create a remote with the standard test fixture url.
+        body = gen_rpm_remote(url=url, policy=policy)
+        remote = self.client.post(RPM_REMOTE_PATH, body)
+        self.addCleanup(self.client.delete, remote['pulp_href'])
+
+        # Sync the repository.
+        self.assertIsNone(repo['latest_version_href'])
+        data = {"repository": repo["pulp_href"]}
+        response = self.client.using_handler(api.json_handler).post(
+            urljoin(remote["pulp_href"], "sync/"), data
+        )
+        sync_task = self.client.get(response["task"])
+        created_at = self.parse_date_from_string(sync_task["pulp_created"])
+        started_at = self.parse_date_from_string(sync_task["started_at"])
+        finished_at = self.parse_date_from_string(sync_task["finished_at"])
+        task_duration = finished_at - started_at
+        waiting_time = started_at - created_at
+        print("\n->     Sync => Waiting time (s): {wait} | Service time (s): {service}".format(
+            wait=waiting_time.total_seconds(),
+            service=task_duration.total_seconds()
+        ))
+
+        repo = self.client.get(repo['pulp_href'])
+        for kickstart_content in get_content(repo)[RPM_KICKSTART_CONTENT_NAME]:
+            self.addCleanup(self.client.delete, kickstart_content['pulp_href'])
+
+        # Check that we have the correct content counts.
+        self.assertIsNotNone(repo['latest_version_href'])
+
+        self.assertIn(
+            RPM_PACKAGE_CONTENT_NAME,
+            get_content_summary(repo).keys(),
+        )
+        self.assertIn(
+            RPM_PACKAGE_CONTENT_NAME,
+            get_added_content_summary(repo).keys(),
+        )
+
+        repo = self.client.get(repo['pulp_href'])
+
+        # Publishing
+        body = {"repository": repo["pulp_href"]}
+        response = self.client.using_handler(api.json_handler).post(RPM_PUBLICATION_PATH, body)
+        publish_task = self.client.get(response["task"])
+        created_at = self.parse_date_from_string(publish_task["pulp_created"])
+        started_at = self.parse_date_from_string(publish_task["started_at"])
+        finished_at = self.parse_date_from_string(publish_task["finished_at"])
+        task_duration = finished_at - started_at
+        waiting_time = started_at - created_at
+        print("\n->     Publish => Waiting time (s): {wait} | Service time (s): {service}".format(
+            wait=waiting_time.total_seconds(),
+            service=task_duration.total_seconds()
+        ))
+
+    def test_epel7_on_demand(self):
```

Comment:
on_demand is relevant to sync only.
Please change the name to "publish" instead.